pontoon/base/static/js/translate.js

```diff
@@ -1109,25 +1109,57 @@ var Pontoon = (function (my) {
       });
 
       // Approve and delete translations
-      $('#helpers .history').on('click', 'menu button', function (e) {
-        var button = $(this);
-        if (button.is('.approve') && button.parents('li.translated').length > 0) {
-          return;
-        }
+      $('#helpers .history').on('click', 'menu .approve', function (e) {
+        var button = $(this),
+            entity = self.getEditorEntity(),
+            translation = $('#translation').val();
 
-        // Approve
-        if (button.is('.approve')) {
-          button.parents('li').click();
+        button.parents('li').click();
 
-          var entity = self.getEditorEntity(),
-              translation = $('#translation').val();
+        // Mark that user approved translation instead of submitting it
+        self.approvedNotSubmitted = true;
+        self.updateOnServer(entity, translation, false, true);
+      });
 
-          // Mark that user approved translation instead of submitting it
-          self.approvedNotSubmitted = true;
-          self.updateOnServer(entity, translation, false, true);
-          return;
-        }
+      $('#helpers .history').on('click', 'menu .unapprove', function (e) {
+         // + send the state of translation to the server 
+         // * update the translation in the editor
+         // * send POST[SAVE] message to the inplace page.
+         // * remove the translated class for the entity.
+         // + update the stats on the whole page
+         var button = $(this);
 
+         $.post('/discard-translation/', {
+            csrfmiddlewaretoken: $('#server').data('csrf'),
+            translation: $(this).parents('li').data('id'),
+            paths: self.getPartPaths(self.currentPart)
+         }).then(function(data) {
+           var entity = self.getEditorEntity(),
+               pf = self.getPluralForm(true);
+
+           var translation = entity.translation[pf];
+
+           self.stats = data.stats;
+           self.authors = data.authors;
+           self.updateProgress();
+           self.updateAuthors();
+           self.updateTranslation(entity, translation, data.translation);
+
+           $('#translation').val('').focus();
+           self.cachedTranslation = '';
+           self.postMessage("SAVE", data.translation);
```

Comment:
This logic needs to change a bit, and act more like deletinga active translations (line 1241).

We need to find the newest translation for the given entity, locale and plural form and return it as `data.translation`. And then set that value into `#translation`, `cachedTranslation` and in-place translation (via `postMessage`).